spec/support/shared_contexts/callback_next_day_check.rb

```diff
@@ -0,0 +1,11 @@
+RSpec.shared_context "callback_next_day_check" do
```

Comment:
Why have 3x shared contexts here instead of just one? It makes it hard to follow what's going on

Reply:
There are 2 shared contexts and 2 shared examples. To test the options method, we need `let(:options_hash) { described_class.options }`, to test the remove_current_day method, we need `let(:options) do
    { Date.today.strftime("%a %d %B") => [
      ["todays times and dates"], ["some more todays times and dates"]
    ],
      Date.tomorrow.strftime("%a %d %B") => [
        ["tomorrows times and dates"], ["some more tomorrows times and dates"]
      ] }
  end
  let(:options_hash) { described_class.remove_current_day(options) }`

Not really a fan of shared contexts myself, you always have to hop around to find out what's going on...

Reply:
Actually, one of those was duplicated, have removed